File: .semgrep/fixtures/no-pii-in-error-logs.py
from app.redaction import redact_tax_identifier_for_log


# ruleid: no-taxpayer-id-in-error-log-py
def log_taxpayer_verification_failure_known_bad(tax_identifier: str, error: Exception) -> None:
    logger.error("Taxpayer verification failed", tax_identifier, error)


# ok: no-taxpayer-id-in-error-log-py
def log_taxpayer_verification_failure_known_good(tax_identifier: str, error: Exception) -> None:
    logger.error(
        "Taxpayer verification failed",
        redact_tax_identifier_for_log(tax_identifier),
        error,
    )

File: .semgrep/fixtures/no-pii-in-error-logs.ts
import { redactTaxIdentifierForLog } from "../../services/tivs-acl/src/redaction.js";

// ruleid: no-taxpayer-id-in-error-log-ts
function logTaxpayerVerificationFailureKnownBad(taxIdentifier: string, error: Error): void {
  logger.error("Taxpayer verification failed", taxIdentifier, error);
}

// ok: no-taxpayer-id-in-error-log-ts
function logTaxpayerVerificationFailureKnownGood(taxIdentifier: string, error: Error): void {
  logger.error("Taxpayer verification failed", redactTaxIdentifierForLog(taxIdentifier), error);
}
